coral-hive/src/main/java/com/linkedin/coral/transformers/CoralRelToSqlNodeConverter.java

```diff
@@ -345,4 +348,41 @@ private SqlNode generateRightChildForSqlJoinWithLateralViews(BiRel e, Result rig
 
     return SqlStdOperatorTable.AS.createCall(POS, asOperands);
   }
+
+  /**
+   * Override this method to handle the conversion for RelNode `f(x).y.z` where `f` is an UDF, which
+   * returns a struct containing field `y`, `y` is also a struct containing field `z`.
+   *
+   * Calcite will convert this RelNode to a SqlIdentifier directly (check
+   * {@link org.apache.calcite.rel.rel2sql.SqlImplementor.Context#toSql(RexProgram, RexNode)}),
+   * which is not aligned with our expectation since we want to apply transformations on `f(x)` with
+   * {@link com.linkedin.coral.common.transformers.SqlCallTransformer}. Therefore, we override this
+   * method to convert `f(x)` to SqlCall, `.` to {@link com.linkedin.coral.common.functions.FunctionFieldReferenceOperator#DOT}
+   * and `y.z` to SqlIdentifier.
+   */
+  @Override
+  public Context aliasContext(Map<String, RelDataType> aliases, boolean qualified) {
```

Comment:
what is the coralSqlNode1 and coralSqlNode2 representation for such SQLs before & after introducing this change?

Reply:
Can you clarify `coralSqlNode1` and `coralSqlNode2`?

Reply:
CoralSqlNode1 is the final sqlNode handed over to calcite for validation & relNode conversion here - https://github.com/linkedin/coral/blob/master/coral-hive/src/main/java/com/linkedin/coral/hive/hive2rel/HiveSqlToRelConverter.java#L57 

CoralSqlNode2 is the output of https://github.com/linkedin/coral/blob/master/coral-hive/src/main/java/com/linkedin/coral/transformers/CoralRelToSqlNodeConverter.java#L51 when using the [hive dialect ](https://github.com/linkedin/coral/blob/master/coral-hive/src/main/java/com/linkedin/coral/transformers/CoralRelToSqlNodeConverter.java#L71)

Reply:
Since it only overrides a method in `CoralRelToSqlNodeConverter`, `coralSqlNode1` is not affected by it.

As for coralSqlNode2, it changes as I mentioned in the comment:
Before override: RelNode `f(x).y.z` will be converted to a SqlIdentifier (like a string), no SqlCall at all
After override: RelNode `f(x).y.z` will be converted to `(f(x).y).z`, where f(x) is a SqlCall (operator is `f`), `f(x).y` is a SqlCall (operator is `.`), `(f(x).y).z` is also a SqlCall (operator is `.`).

Reply:
I meant, for a given Hive SQL - "select f(x).y.z from foo", what is the `coralSqlNode1` we construct right now? and is the code change making sure that the `coralSqlNode2` we build is same as `coralSqlNode1` from LHS?

Lets try to add minimum and only necessary overrides in CoralRelToSqlNodeConverter

Also, Is there a UT that captures this corner case?

Reply:
1. Yes, for `select f(x).y.z from foo`, the current `coralSqlNode1` is `(f(x).y).z` (all SqlCalls instead of SqlIdentifier), so the code change makes sure that the `coralSqlNode2` we build is same as `coralSqlNode1` from LHS. 
2. I think it is already the minimum override for handling this corner case.
3. No UT for now, but I will add it. Would like to address comments together. Please review and comment if you have concerns on other parts.

Reply:
Please add a UT for this behavior override.